src/social-network-provider/twitter.com/ui/fetch.ts

```diff
@@ -0,0 +1,131 @@
+import {
+    bioCard,
+    fromPostSelectorsSelectPostContentString,
+    postPopupSelector,
+    postsContentSelectors,
+    postsRootSelector,
+    postsSelectors,
+    selfInfoSelectors,
+} from '../utils/selector'
+import { MutationObserverWatcher } from '@holoflows/kit'
+import Services from '../../../extension/service'
+import { PersonIdentifier } from '../../../database/type'
+import { host } from '../index'
+import { getEmptyPostInfo, SocialNetworkUI } from '../../../social-network/ui'
+import { deconstructPayload } from '../../../utils/type-transform/Payload'
+import { regexMatch, timeout } from '../../../utils/utils'
+import { hasPostPopup } from '../utils/status'
+
+export const resolveLastRecognizedIdentity = (self: SocialNetworkUI) => {
+    const selfSelector = selfInfoSelectors.screenName
+    const assign = () => {
+        const ref = self.lastRecognizedIdentity
+        const i = new PersonIdentifier(host, selfInfoSelectors.screenName.evaluateOnce()[0])
+        if (i.isUnknown) throw new Error('failed to recognize user id')
```

Comment:
This function expects to fail to collect info, maybe don't thorw is a good idea

Reply:
a06055dc